source/Core/BSP/Pinecilv2/ble_handlers.cpp

```diff
@@ -244,15 +244,19 @@ int ble_char_write_setting_value_callback(struct bt_conn *conn, const struct bt_
       }
     } else if (uuid_value < SettingsOptions::SettingsOptionsLength) {
       setSettingValue((SettingsOptions)(uuid_value), new_value);
-      // @TODO refactor to make this more usable
-      if (uuid_value == SettingsOptions::OLEDInversion) {
+      switch (uuid_value) {
+      case SettingsOptions::OLEDInversion:
         OLED::setInverseDisplay(getSettingValue(SettingsOptions::OLEDInversion));
-      }
-      if (uuid_value == SettingsOptions::OLEDBrightness){
+        break;
+      case SettingsOptions::OLEDBrightness:
         OLED::setBrightness(getSettingValue(SettingsOptions::OLEDBrightness));
-      }
-      if (uuid_value == SettingsOptions::OrientationMode){
+        break;
+      case SettingsOptions::OrientationMode:
         OLED::setRotation(getSettingValue(SettingsOptions::OrientationMode) & 1);
+        break;
+        default:
+          BT_ASSERT(uuid_value == uuid_value);
```

Comment:
Why this assert?

Reply:
the intention was to notify that this code path was reached. In general it should mean that this block needs to be revisited (or something went really wrong) - since these are currently the only parameters which require additional handling to apply them without restart.

Reply:
I'm not sure why this would be used; this will always evaluate to `true` and thus not crash the chip.

Reply:
I don't disagree. should we crash the chip?

we left it this way to ask you what should be done, basically.

Reply:
I think for now we remove this; but I'm going to have to cleanup an "apply" function for settings